logrotate.c

```diff
@@ -2554,6 +2554,11 @@ int main(int argc, const char **argv)
     logSetLevel(MESS_NORMAL);
     setlocale (LC_ALL, "");
 
+    /* Creates the dir for state file and moves old
+       state file to new location */
+    mkdir(STATEDIR, 0755);
+    rename(OLDSTATEFILE, STATEFILE);
+
```

Comment:
oh, and by FHS definition `/var/run` is wrong path to store state, `run` is wiped on reboot, this should be `/var/lib` which is not reset on reboots.

that being said, i still would not change current behaviour in logrotate upstream. leave existing path whatever it was not to bring unexpected incompatibilities in patch release.